static/js/src/publisher-admin/pages/Releases/AddTrackPanel.tsx

```diff
@@ -0,0 +1,178 @@
+import {
+  Accordion,
+  Button,
+  Input,
+  Panel,
+  Spinner,
+} from "@canonical/react-components";
+import { useState } from "react";
+import { useRecoilState } from "recoil";
+import { packageDataState } from "../../state/atoms";
+import { usePackage } from "../../hooks";
+
+type AddTrackPanelProps = {
+  charmName: string;
+  onClose: () => void;
+  setSelectedTrack: (track: string) => void;
+};
+
+export default function AddTrackPanel({
+  charmName,
+  onClose,
+  setSelectedTrack,
+}: AddTrackPanelProps) {
+  const [trackName, setTrackName] = useState("");
+  const [versionPattern, setVersionPattern] = useState("");
+  const [phasingPercentage, setPhasingPercentage] = useState("");
+  const { refetch } = usePackage(charmName);
+
+  const [loading, setLoading] = useState(false);
+  const [errors, setErrors] = useState<{ [key: string]: string | undefined }>(
+    {}
+  );
+
+  const submit = async () => {
+    if (!trackName || !!Object.keys(errors).filter((key) => errors[key]).length)
+      return;
+
+    setLoading(true);
+
+    const formData = new FormData();
+    formData.append("csrf_token", window.CSRF_TOKEN);
+    formData.append("track-name", trackName);
+    if (versionPattern) formData.append("version-pattern", versionPattern);
+    if (phasingPercentage)
+      formData.append("auto-phasing-percentage", phasingPercentage);
+
+    const response = await fetch(`/${charmName}/create-track`, {
+      method: "POST",
+      body: formData,
+    });
+
+    const responseData = await response.json();
+
+    if (response.ok) {
+      refetch();
+      setSelectedTrack(trackName);
```

Comment:
On successful addition of a new track, can you close the side panel, switch to the new track and also show a success message like we do in Snapcraft.io? This should make it very clear to the user that the track has been created successfully